common/src/main/java/com/wynntils/services/mapdata/attributes/impl/FadingMapVisiblity.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.mapdata.attributes.impl;
+
+import com.wynntils.services.mapdata.attributes.type.MapVisibility;
+
+public class FadingMapVisiblity implements MapVisibility {
+    private final float min;
+    private final float max;
+    private final float fade;
+
+    public FadingMapVisiblity(float min, float max, float fade) {
+        this.min = min;
+        this.max = max;
+        this.fade = fade;
+    }
+
+    @Override
+    public float getVisibility(float zoomLevel) {
+        // A feature stats fading in at min - fade, is fully visible at min,
```

Comment:
I'd like to see this somewhat different. The change I would like to see is that the fading should be "centered" around min and max, instead of extending the visibility. That is, if you change the fade it should still feel like the icon is shown for roughly the same interval. With this code, increasing the fade will make the icon visibile for a much longer period, and that will likely make people set a shorter visible interval to get the desired effect.

So basically what I am saying is that the max/min specifies the interval where the icon is shown with at least 50% visibility.

Reply:
Is that desirable behavior, though? Personally, I would expect min max values a point where they either start or stop fading. Now, it's a mix of both, which I agree can be a bit confusing, but overall, I thought this makes the most sense, without having to read "documentation" for the user.

Reply:
I think that is desirable, yes. Here are my two arguments:

1) This decouples fading from the actual max/min point. Otherwise you will probably need to adjust max/min at the same time you adjusts fading, to get your desired effect.

2) If you don't select the midpoint of the fading range, then what do you select? When the point starts to fade or stops to fade? It is kind of impossible to know.

My idea is to think of this as without fading. Then there is a single instance at which the visibility kicks in (or out). But the `fade` argument makes this a little less abrupt. But if you slide the fading up or down, it should still center around the actual value.